### ISSUE TITLE
Visual Composer Shortcodes showing in product description :(

### ISSUE THREAD

**Issue by Adrianart** [facebookincubator/facebook-for-woocommerce#547]
Hi. I have a problem with the shortcodes showing in the product description of the product.
I've tried to add this to the function _clean_string_ in the _fbutils.php_ file, **before** return $string; : 
```
$shotcodes_tags = array( 'vc_row', 'vc_column', 'vc_column', 'vc_column_text', 'vc_message' );
$string = preg_replace( '/\[(\/?(' . implode( '|', $shotcodes_tags ) . ').*?(?=\]))\]/', '', $string );
```
but did not work.
Exemple of shortcodes to strip: [vc_row full_width='stretch_row_content_no_spaces' gap='30' content_placement='middle' css='.vc_custom_1529240876803{background-color: #ffffff !important;}'][vc_column][vc_row_inner content_placement='middle' gap='30'][vc_column_inner width='1/6'][/vc_column_inner][vc_column_inner width='2/6'].
Thanks

Thanks!

Plugin: 1.9.0
WC: 3.4.3
WP: 4.9.6
PHP: 7.2.5
Website: https://shop.artivist.ro
FB page: https://www.facebook.com/shop.artivist.ro/
Screenshot URL: https://www.facebook.com/commerce/products/2438710926146421/
![fb](https://user-images.githubusercontent.com/20039859/41717035-148df48a-7561-11e8-8fe4-aa27eae19281.jpg)

**Comment by dmitridr** [comment]
## Automated Support Message

Hello,

Thanks for your interest in our plugin!

If your issue is a feature request or feedback for the developers, reply with `Feature Request` or `Feedback`.
If you are a developer or have a question for the developers related to code, reply with `Developer Question`.
Your comment will be categorized accordingly, and we will read it and reply when we can.

If you have encountered a bug or a problem with the plugin, the steps below contain information to help you resolve the issue yourself and provide us information we need to help you with your problem.

## Search Existing Issues

**First, please [search our issues](https://github.com/facebookincubator/facebook-for-woocommerce/issues?q=) to see if your problem has been reported before.**
There may already be a workaround or a solution to your problem. You can search our issues using [the search bar near the top of our issues page](https://github.com/facebookincubator/facebook-for-woocommerce/issues?q=); be sure to search both _open_ and _closed_ issues.
If you do find a similar issue, leave a comment on that issue and close this one. Your comment helps us prioritize which existing issues are important and urgent.

## Common Resolution Methods

**If you are experiencing a problem or bug and have not found any similar issue, please try the following:**

* Upgrade to the latest version of the plugin.
* Try to `Resync` products using the button on the plugin config page
* Reset the plugin by selecting `Reconfigure > Advanced Options > Delete`, and setting up this plugin again.
* Try to deactivate other plugins that you think may be interfering, then reset and set up again.

## Necessary Information

**If the above steps did not help, please provide the following information in your next comment in order for us to debug your problem:**

* WooCommerce version
* WordPress version
* Plugin version (please note: we only support up-to-date versions of the plugin)
* A screenshot of where you see the problem (for example: if it is on Facebook, a screenshot of Facebook, if it is on WooCommerce, a screenshot of WooCommerce. If it is in multiple places, multiple screenshots.)
* The URL of your website
* The URL of your page
* The URL(s) of where you took your screenshot(s)
* (If applicable) Detailed steps you took in order to see the problem

Please include `Debug Ready` in your reply so our automated system can flag your issue for manual review. Please note that due to a high volume of issues, we may not be able to respond immediately.

Thanks for your patience,
The Facebook Dev Team (@mengyingdu, @dmitridr, @sammul, @jieminz22)

**Comment by Adrianart** [comment]
Debug Ready
![fb](https://user-images.githubusercontent.com/20039859/41766059-9f215a76-760e-11e8-8d7e-d4bea84f8943.jpg)

Plugin: 1.9.0
WC: 3.4.3
WP: 4.9.6
PHP: 7.2.5
Website: https://shop.artivist.ro
FB page: https://www.facebook.com/shop.artivist.ro/
Screenshot URL: https://www.facebook.com/commerce/products/2438710926146421/

**Comment by dmitridr** [comment]
Hello,

Thanks for reporting this, looks like you are very close to solving the issue yourself. We actually have a filter you can use to easily do edits to the product data, including description. 

The filter is described in #48 and a couple of other issues. 
Try using your code with the filter, so the code would be something like this:

```
add_filter( 'facebook_for_woocommerce_integration_prepare_product', 'wc_fb_product_fix', 10, 2);
function wc_product_fix($product, $id){
  $shotcodes_tags = array( 'vc_row', 'vc_column', 'vc_column', 'vc_column_text', 'vc_message' );
  $product['description'] = preg_replace( '/\[(\/?(' . implode( '|', $shotcodes_tags ) . ').*?(?=\]))\]/', '', 
  $product['description']);
  return $product;
}
```

This should work, assuming your regular expression is correct.

**Comment by dmitridr** [comment]
I think

```
$product['description'] = preg_replace( '\[vc_[^\]]*\]', '', $product['description']);
```

Could be a simpler version of what you want?

**Comment by dmitridr** [comment]
I tested regular expressions locally to try and get a simpler version of what you wanted.
This should work:

```
$product['description'] = preg_replace( '/\[.{0,1}vc_.*?\]/', '', $product['description']);
```

**Comment by Adrianart** [comment]
Dmitri D , you're a wizard! It worked.
Thank you very much!

**Comment by dmitridr** [comment]
You're welcome, glad I could help! 
One thing to note is if you inserted this code in the plugin, every time you update the plugin, you'll have to re-insert the code. In order to avoid this, the code can be inserted in another plugin or in `functions.php` (there are some guides online about how this can be done). 

Please consider giving us a review at https://woocommerce.com/products/facebook/ if you find our plugin useful! Closing this out.

**Comment by eylulgormus** [comment]
I have the same problem. I tried adding the code above to my child theme's functions.php but it didn't solve the issue. Here's a screenshot showing the problem. Also, in the product editing page on woocommerce, there's a "Facebook description" section. I tried adding plain text to that area hoping Facebook would fetch that data but that didn't work either.

![screen shot 2018-11-07 at 13 42 47](https://user-images.githubusercontent.com/42125195/48126758-0fec1480-e293-11e8-9a4c-2d5690752b21.png)
